docs/supervisor/development.md

```diff
@@ -21,6 +21,73 @@ The instructions here is for development of the Supervisor itself.
+
+:::tip
+You must use WSL2. WSL version 1 is not capable of running Docker and therefore is incompatible. If using this method, you must ensure you have [set up WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+:::
+
+It is assumed at this point you have installed Debian from the Microsoft Store and [Visual Studio Code](https://code.visualstudio.com/download). From within your WSL2-enabled Debian terminal, run the following annotated bash script. If you copy the entire script it should perform the required setup of Debian for Home Assistant development but any problems may be difficult to troubleshoot, so it is recommended to run this line-by-line.
+
+```bash
+apt update; apt install -y curl wget git gnupg; #Update and install our dependencies
+curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash; #Install NVM
+source ~/.profile; nvm install node; #reload profile and install node
+sudo apt remove cmdtest yarn  #Remove problematic packages if installed. Debian current Yarn does not work.
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -  #Add yarn repo certificate
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list #Add Yarn repo
+sudo apt update; sudo apt install yarn -y; #Install yarn
+curl -fsSL https://get.docker.com |sudo bash  #install docker Community Edition
+sudo update-alternatives --set iptables /usr/sbin/iptables-legacy #Fix IP Tables for Docker
+git clone https://github.com/home-assistant/supervisor /home/$USER/supervisor #Clone Supervisor to ~/supervisor
+usermod -aG docker $USER #Add self to docker group
+systemctl enable docker.service #enable docker start on boot
+service docker start #Start the docker service
+service docker status #Verify docker status is running
+```
+
+Restart your Debian terminal to activate the new profiles and software configurations. 
+
+:::tip
+Always run scripts line by line to figure out where the problem lies. If you need to completely start over, you can uninstall the Debian app, then from the Windows command prompt execute `del %HOMEPATH%\AppData\Roaming\Code`. After removing your VSCode settings, launch VS Code, and remove all extensions. Finally either execute `net stop LxssManager` from an administrative `cmd` prompt or reboot your computer and try again.
+:::
+
+Now you're ready to run Yarn and Node. We must set up Docker communications and launch Visual Studio Code. The following commands will need to be run after each reboot/restart to link Docker to your WSL instance and launch Visual Studio Code.
+
+```bash
+sudo mkdir /sys/fs/cgroup/systemd
+sudo mount -t cgroup -o none,name=systemd cgroup /sys/fs/cgroup/systemd
+sudo service docker start
+code
+```
```

Comment:
If stuff like this is needed, this instruction should really not use the standard wsl instance since it will break normal wsl uses.

Reply:
This was required for wsl1

Reply:
Planning to do an integration coming soon. I'll be able to update for WSL 2